gradle-junit-reports/src/test/java/com/palantir/gradle/junit/BuildJunitReportServiceTests.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * (c) Copyright 2017 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.gradle.junit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import com.palantir.gradle.junit.Report.TestCase;
+import org.gradle.internal.build.event.types.DefaultTaskDescriptor;
+import org.gradle.tooling.events.internal.DefaultOperationDescriptor;
+import org.gradle.tooling.events.task.TaskExecutionResult;
+import org.gradle.tooling.events.task.TaskFinishEvent;
+import org.gradle.tooling.events.task.internal.DefaultTaskFailureResult;
+import org.gradle.tooling.events.task.internal.DefaultTaskFinishEvent;
+import org.gradle.tooling.events.task.internal.DefaultTaskOperationDescriptor;
+import org.gradle.tooling.events.task.internal.DefaultTaskSuccessResult;
```

Comment:
this is where I'd like advice. Internal gradle imports are not allowed... I suppose I could mock out the types? I could also add checkstyle suppressions. That's the main reason the build's failing, anyhow.